sycl/source/detail/device_image_impl.hpp

```diff
@@ -427,6 +1174,13 @@ class device_image_impl {
   // Contains map of spec const names to their descriptions + offsets in
   // the MSpecConstsBlob
   std::map<std::string, std::vector<SpecConstDescT>> MSpecConstSymMap;
+
+  // MOrigins is a bitfield to allow cases where the image is the product of
+  // merging images of different origins.
+  uint8_t MOrigins = ImageOriginSYCLOffline;
```

Comment:
That seems to correlate with `std::variant`'s state above, at least partially. Why do we need both?

Reply:
As we build/compile/link images, the variant will change, but the origins get merged. I.e. if you build a SYCLBIN the variant will end up containing a binary image, but the origin will be `ImageOriginSYCLBIN`. If you link a SYCLBIN and an RTC compiled binary, it will be a binary image and `ImageOriginRTC | ImageOriginSYCLBIN`.